libraries/chain/steem_evaluator.cpp

```diff
@@ -2335,19 +2357,61 @@ void claim_account_evaluator::do_apply( const claim_account_operation& o )
 
    FC_ASSERT( creator.balance >= o.fee, "Insufficient balance to create account.", ( "creator.balance", creator.balance )( "required", o.fee ) );
 
-   const auto& creation_fee = wso.median_props.account_creation_fee;
-
-   asset paid_fee = o.fee;
-
-   if( paid_fee < creation_fee )
-   {
-      // This is optimized because STEEM_100_PERCENT == STEEM_ACCOUNT_SUBSIDY_PRECISION
-      uint64_t percent_unpaid = ( ( creation_fee.amount.value - paid_fee.amount.value ) * STEEM_100_PERCENT ) / creation_fee.amount.value;
+   FC_TODO( "Remove min() after HF20" );
+   int64_t creation_fee = std::min( wso.median_props.account_creation_fee.amount.value, STEEM_MAX_ACCOUNT_CREATION_FEE );
+
+   FC_ASSERT( o.fee.amount.value <= creation_fee, "Cannot pay more than account creation fee. paid: ${p} fee: ${f}", ("p", o.fee.amount.value)("f", creation_fee) );
+
+   int64_t fee_discount = creation_fee - o.fee.amount.value;
+
+   if( fee_discount > 0 )
```

Comment:
I thought the intent behind this change was to make it so the the fee was wither the creation fee or it was 0.